edge/pkg/edged/edged_pods.go

```diff
@@ -1150,3 +1151,165 @@ func (e *edged) getHostIPByInterface() (string, error) {
 	}
 	return "", fmt.Errorf("no ip and mask in this network card")
 }
+
+// findContainer finds and returns the container with the given pod ID, full name, and container name.
+// It returns nil if not found.
+func (e *edged) findContainer(podFullName string, podUID types.UID, containerName string) (*kubecontainer.Container, error) {
+	pods, err := e.containerRuntime.GetPods(false)
+	if err != nil {
+		return nil, err
+	}
+	// Resolve and type convert back again.
+	// We need the static pod UID but the kubecontainer API works with types.UID.
+	podUID = types.UID(e.podManager.TranslatePodUID(podUID))
+	pod := kubecontainer.Pods(pods).FindPod(podFullName, podUID)
+	return pod.FindContainerByName(containerName), nil
+}
+
+// RunInContainer runs a command in a container, returns the combined stdout, stderr as an array of bytes
+func (e *edged) RunInContainer(podFullName string, podUID types.UID, containerName string, cmd []string) ([]byte, error) {
```

Comment:
where does this method got called?

Reply:
Need implement the HostInterface interface in edged for the handler.